File: proyectomds2/src/main/java/vistas/VistaElementoBuscadorAlbum.java
package vistas;

import com.vaadin.flow.templatemodel.TemplateModel;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.dom.Element;

/**
 * A Designer generated component for the vista-elemento-buscador-album template.
 *
 * Designer will add and remove fields with @Id mappings but
 * does not overwrite or otherwise change this file.
 */
@Tag("vista-elemento-buscador-album")
@JsModule("./src/vista-elemento-buscador-album.js")
public class VistaElementoBuscadorAlbum extends PolymerTemplate<VistaElementoBuscadorAlbum.VistaElementoBuscadorAlbumModel> {

    @Id("vaadinButton")
	private Button vaadinButton;
	@Id("autor_album")
	private Label autor_album;
	@Id("vaadinHorizontalLayout")
	private HorizontalLayout vaadinHorizontalLayout;
	@Id("vaadinVerticalLayout")
	private Element vaadinVerticalLayout;

	/**
     * Creates a new VistaElementoBuscadorAlbum.
     */
    public VistaElementoBuscadorAlbum() {
        // You can initialise any data required for the connected UI components here.
    }

    /**
     * This model binds properties between VistaElementoBuscadorAlbum and vista-elemento-buscador-album
     */
    public interface VistaElementoBuscadorAlbumModel extends TemplateModel {
        // Add setters and getters for template properties here.
    }
}

File: proyectomds2/frontend/src/vista-elemento-buscador-album.js
import {html, PolymerElement} from '@polymer/polymer/polymer-element.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-horizontal-layout.js';
import '@vaadin/vaadin-button/src/vaadin-button.js';

class VistaElementoBuscadorAlbum extends PolymerElement {

    static get template() {
        return html`
<style include="shared-styles">
                :host {
                    display: block;
                    height: 100%;
                }
            </style>
<vaadin-vertical-layout theme="spacing" style="width: 100%; height: 100%;" id="vaadinVerticalLayout">
 <vaadin-horizontal-layout theme="spacing" style="width: 100%;" id="vaadinHorizontalLayout">
  <vaadin-button style="width: 100%;" id="vaadinButton" dir="album_autor">
    Button 
  </vaadin-button>
  <label style="width: 100%; align-self: center; flex-grow: 1;" id="autor_album">Autor</label>
 </vaadin-horizontal-layout>
</vaadin-vertical-layout>
`;
    }

    static get is() {
        return 'vista-elemento-buscador-album';
    }

    static get properties() {
        return {
            // Declare your properties here.
        };
    }
}

customElements.define(VistaElementoBuscadorAlbum.is, VistaElementoBuscadorAlbum);
